src/pages/Cart/cartFn.jsx

```diff
@@ -1,23 +1,72 @@
-
-
-export function cartFn(state,{type,payload,initialQty}){
-    console.log("payload",{payload,type});
-    switch(type){
-        case "INITIAL":
-            console.log("initial payload",payload ,"first element type","reducer",[...payload].reduce((sum,num)=>({...sum,price:Number(sum.price)+Number(num.price)}),{price:0}));
-            return{...state,
-                subTotal: payload.length>1?[...payload].reduce((sum,num)=>({...sum,price:Number(sum.price)+Number(num.price)}),{price:0}):payload.length>0?payload[0]:0,
-                }
-        case "SUBTRACT":
-            console.log("subtract payload",payload,"length of payload", payload.length);
-            return{...state,
-                subTotal:payload.length>0?[...payload].reduce((sum,num)=>({...sum,price:Number(sum.price)*sum.qty+Number(num.price)*num.qty}),{price:0,qty:1}):0
-            }
-        case "INCREASE":
-            console.log("inccrease payload",payload,"initial qty",initialQty);
-            return{...state,
-                subTotal:payload.length>0?[...payload].reduce((sum,num)=>({...sum,price:Number(sum.price)*sum.qty+Number(num.price)*num.qty}),{price:0,qty:1}):0,
-                qty:initialQty
-            }
-    }
-}
\ No newline at end of file
+export function cartFn(state, { type, payload, initialQty }) {
+  console.log("payload", { payload, type });
+  switch (type) {
+    case "INITIAL":
+      console.log(
+        "initial payload",
+        payload,
+        "first element type",
+        "reducer",
+        [...payload].reduce(
```

Comment:
I dont think you require a reducer for this functionality. My understanding is that since these computations are based on your existing cartState, you can do away without using a reducer here. Generally speaking, computed data shouldn't be stored in the state, instead it should be computed as needed as part of the render method.